From 5e9b538db5304ccfed6c0d048a94d8c47dd27e31 Mon Sep 17 00:00:00 2001
From: Jeremy Bicha <jeremy.bicha@canonical.com>
Date: Thu, 17 Feb 2022 23:51:15 +0100
Subject: [PATCH] games: Only use -Ono-letrectify with guile 3.0

This fixes a build failure on guile 2.2 where this flag is unknown.

Fixes: https://gitlab.gnome.org/GNOME/aisleriot/-/issues/925
---
 games/meson.build | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/games/meson.build b/games/meson.build
index e552beab..56819b4d 100644
--- a/games/meson.build
+++ b/games/meson.build
@@ -122,9 +122,13 @@ guile_warnings = [
   '-W1',
 ]
 
-guile_flags = [
-  '-Ono-letrectify',
-]
+guile_flags = []
+
+if guile_version.version_compare('>= 3.0')
+  guile_flags += [
+    '-Ono-letrectify',
+  ]
+endif
 
 # FIXME: this should use files('...') (allowed as per docs),
 # but that crashes meson, see https://gitlab.gnome.org/chpe/meson/issues/2
-- 
GitLab

